src/GameplaySystem.cpp

```diff
@@ -401,6 +402,26 @@ void GameplaySystem::handle_object_transform_event(const Event& e) {
             )
         );
     }
+
+    glm::vec3 it_pos = glm::vec3(
+                           object_positions_[current_it_id_].x,
+                           object_positions_[current_it_id_].y,
+                           object_positions_[current_it_id_].z
+                       );
+
+    glm::vec3 vec_to = glm::normalize(glm::vec3(it_pos - glm::vec3(x, y, z)));
+    glm::mat4 rot_matrix = glm::toMat4(glm::quat(qw, qx, qy, qz));
+    glm::vec4 vec_to_it = glm::vec4(vec_to, 0.f) * rot_matrix;
```

Comment:
This would give you incorrect results. Matrix should go on the left. Also, you want the 4th component of that vec4 to be 1.0f.

```
glm::vec4 vec_to_it = rot_matrix * glm::vec4(vec_to, 1.0f);
```